### PR TITLE
create referral forms refactoring based on the business feedback

### PR DESCRIPTION
**Pull request by jasonfungsing**

### CODE REVIEW

**Comment by nicjb** [comment]
Hey @jasonfungsing - from my reading it looks like you are indicating mandatory fields with a "*" at the end, is that right? If so, we need to flip that around as per QA1-002 to have "(optional)" at the end of the optional fields. No indicator on mandatory fields - just the lack of "(optional)"

File: frontend/apps/orams/actions/referralActions.js [hunk 1]
```diff
@@ -56,10 +56,9 @@ export const setDelegateInformationFormStatus = data => dispatch => {
 export const createReferral = data => dispatch => {
   dispatch(sendingRequest(true))
 
-  // todo need work here
   const referralData = {
     ...data,
-    referralType: 'Compensation',
+    referralType: data.referralDetails.is_compensation_case ? 'Compensation' : 'Non Compensation',
```
Comment (bsigney) [inline frontend/apps/orams/actions/referralActions.js]: @jasonfungsing where is is_compensation_case getting set in the form?  I don't see the field.

These data need to match the enums in the database.

Also why does it need to be boolean?  Couldn't it be a straight select input with two options (Compensation and Non-Compensation)

Comment (jasonfungsing) [reply frontend/apps/orams/actions/referralActions.js]: done

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js [hunk 2]
```diff
@@ -47,14 +44,94 @@ class ReferralBuilderServiceDetailsForm extends BaseForm {
...
             onSubmit={data => handleSubmit(data)}
           >
+            <div className="field">
+              <label htmlFor="is_compensation_case">
+                <b>Is this a compensation (Comcare) referral? *</b>
+              </label>
+              <Select
+                block
+                onChange={e => this.onChangeDropdown('is_compensation_case', e)}
+                model={`${model}.is_compensation_case`}
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js]: rename `is_compensation_case` as this is no longer a boolean.

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js [hunk 3]
```diff
@@ -47,14 +44,94 @@ class ReferralBuilderServiceDetailsForm extends BaseForm {
...
+                model={`${model}.is_liability_determined`}
+                id="is_liability_determined"
+                name="is_liability_determined"
+                options={[
+                  {
+                    value: ``,
+                    label: ' '
+                  },
+                  {
+                    value: `true`,
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js]: does this get saved as boolean or string in the back-end?
view referral is expecting a boolean - need to modify if this is not the case.

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js [hunk 4]
```diff
@@ -66,247 +143,379 @@ class ReferralBuilderServiceDetailsForm extends BaseForm {
...
+                model={`${model}.is_employee_currently_at_work`}
+                id="is_employee_currently_at_work"
+                name="is_employee_currently_at_work"
+                options={[
+                  {
+                    value: ``,
+                    label: ' '
+                  },
+                  {
+                    value: 'yes',
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js]: view referral is expecting a boolean - need to modify if this is not the case.

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js [hunk 5]
```diff
@@ -66,247 +143,379 @@ class ReferralBuilderServiceDetailsForm extends BaseForm {
...
+                  {
+                    value: `same`,
+                    label: 'Same'
+                  },
+                  {
+                    value: `reduced`,
+                    label: 'Reduced'
+                  },
+                  {
+                    value: `altered_hours`,
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js]: this does not need to be snake-cased. can just save it as 'Altered Hours' on back-end.
View referrals would just need to output this case

same goes for all options in this dropdown

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js [hunk 6]
```diff
@@ -66,247 +143,379 @@ class ReferralBuilderServiceDetailsForm extends BaseForm {
...
+                model={`${model}.post_injury_work_duties`}
+                name="post_injury_work_duties"
+                id="post_injury_work_duties"
+                options={[
+                  {
+                    value: `same`,
+                    label: 'Same'
+                  },
+                  {
+                    value: `modified_duties`,
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderServiceDetailsForm.js]: this does not need to be snake-cased. can just save it as 'Modified Duties' on back-end.
View referrals would just need to output this case

same goes for all options in this dropdown

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderPersonalDetailsForm.js [hunk 7]
```diff
@@ -60,100 +56,289 @@ class ReferralBuilderPersonalDetailsForm extends BaseForm {
...
+                  {
+                    value: `female`,
+                    label: 'Female'
+                  },
+                  {
+                    value: `male`,
+                    label: 'Male'
+                  },
+                  {
+                    value: `transgender_female`,
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderPersonalDetailsForm.js]: this does not need to be snake-cased. can just save it as 'transgender female' on back-end.
View referrals would just need to output this case

same goes for all options in this dropdown

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderPersonalDetailsForm.js [hunk 8]
```diff
@@ -60,100 +56,289 @@ class ReferralBuilderPersonalDetailsForm extends BaseForm {
...
+                name="employee_current_employment_type"
+                id="employee_current_employment_type"
+                label="Employee current employment type"
+                options={[
+                  {
+                    value: ``,
+                    label: ''
+                  },
+                  {
+                    value: `full_time`,
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilderPersonalDetailsForm.js]: this does not need to be snake-cased. can just save it as full time' on back-end.
View referrals would just need to output this case

same goes for all options in this dropdown

**Pull request merged**